python/ray/tests/test_basic.py

```diff
@@ -864,6 +864,40 @@ def method(self):
     assert ray.get([id1, id2, id3, id4]) == [0, 1, "test", 2]
 
 
+def test_many_fractional_resources(shutdown_only):
```

Comment:
Can we augment this test so that `f` takes in a bool which says whether to `block` or not, and if block is true then it does something like `ray.get(g.remote())`.

This will hit the code path where tasks release there CPU resources and then reacquire them, which I think we currently aren't testing.